arangod/Cluster/ClusterComm.cpp

```diff
@@ -1149,7 +1158,11 @@ ClusterCommThread::ClusterCommThread() : Thread("ClusterComm"), _cc(nullptr) {
   _communicator = std::make_shared<communicator::Communicator>();
 }
 
-ClusterCommThread::~ClusterCommThread() { shutdown(); }
+void ClusterCommThread::haltThreads() {
+  shutdown();
+}
+
+ClusterCommThread::~ClusterCommThread() { }
```

Comment:
Please add a comment why in this specific case we do not want to call shutdown in the destructor, and add an assert in the destructor that `state() == ThreadState::STOPPED`.